### PR TITLE
fix(auth): resolve token refresh loop and preserve ModelStates on auth reload

### LINKED ISSUE #797: Bug: ModelStates (BackoffLevel) lost when auth is reloaded or refreshed
## Bug Description

When an auth file is modified/reloaded (triggered by file system watcher) or when auth token is refreshed, the `ModelStates` field (containing `BackoffLevel` for quota cooldown) is lost. This causes the exponential backoff to reset to 0 on every 429 error.

### Issue 1: `service.go:applyCoreAuthAddOrUpdate`
Only preserves `CreatedAt`, `LastRefreshedAt`, `NextRefreshAfter` - missing `ModelStates`.

### Issue 2: `conductor.go:Manager.Update`
Only preserves `Index`, `indexAssigned` - missing `ModelStates`.

Since `ModelStates` is only in memory (not persisted to JSON), it gets lost when auth is replaced.

## Impact

1. 429 error → `persist()` writes auth file
2. File watcher triggers `processing incrementally`
3. Auth reloaded without `ModelStates`
4. `BackoffLevel` resets to 0
5. Next 429 gets 1s cooldown instead of exponentially increasing

Results in 90+ failures instead of reaching 30-min max cooldown after ~12 failures.

## Suggested Fix

Preserve `ModelStates` in both locations:

```go
if len(auth.ModelStates) == 0 && len(existing.ModelStates) > 0 {
    auth.ModelStates = existing.ModelStates
}
```

## Additional Enhancement

Parse upstream `retryDelay` from Antigravity 429 responses:

```json
{"@type": "type.googleapis.com/google.rpc.RetryInfo", "retryDelay": "10627.493230411s"}
```

Use this (~3 hours) instead of default exponential backoff.

### LINKED ISSUE #833: [Bug] Antigravity token refresh loop caused by metadataEqualIgnoringTimestamps skipping critical field updates
**Is it a request payload issue?**
[  ] Yes, this is a request payload issue. I am using a client/cURL to send a request payload, but I received an unexpected error.
[x] No, it's another issue.

**Describe the bug**

Antigravity OAuth tokens are being refreshed every 5 seconds in an infinite loop. The root cause is that `metadataEqualIgnoringTimestamps()` in `sdk/auth/filestore.go` intentionally ignores `access_token`, `expired`, `timestamp`, and `expires_in` fields when comparing metadata. This causes the `Save()` function to skip writing these critical fields to disk.

As a result:
1. Token is refreshed successfully in memory
2. `Save()` compares old vs new metadata, ignoring token-related fields
3. Since other fields (email, refresh_token, type) haven't changed, write is skipped
4. The `expired` field in the file remains stale (e.g., days ago)
5. Next refresh check (every 5 seconds) sees expired token and triggers refresh again
6. Infinite loop continues

**Evidence from auth file:**
```json
{
    "expired": "2025-01-01T10:00:00Z",  // ← Example: shows a date in the past
    "timestamp": 1735689600000,          // ← Example: not updated after refresh
    "expires_in": 3599
}
```
File modification time shows current time, but `expired` and `timestamp` fields were not actually updated.

**Problematic code** (`sdk/auth/filestore.go:287-307`):
```go
func metadataEqualIgnoringTimestamps(a, b []byte) bool {
    // ...
    // These fields are ignored, causing the bug:
    ignoredFields := []string{"timestamp", "expired", "expires_in", "last_refresh", "access_token"}
    for _, field := range ignoredFields {
        delete(objA, field)
        delete(objB, field)
    }
    return deepEqualJSON(objA, objB)
}
```

When this returns `true`, the write is skipped:
```go
if metadataEqualIgnoringTimestamps(existing, raw) {
    return path, nil  // Skip write!
}
```

**Suggested fix:**

Option 1: Always write when `access_token` changes
```go
func shouldSkipWrite(existing, new []byte) bool {
    var objA, objB map[string]any
    json.Unmarshal(existing, &objA)
    json.Unmarshal(new, &objB)
    
    // Must write if access_token changed
    if objA["access_token"] != objB["access_token"] {
        return false
    }
    // Can skip for other changes
    return metadataEqualIgnoringTimestamps(existing, new)
}
```

Option 2: Remove `access_token`, `expired`, `timestamp` from ignored fields - only ignore `last_refresh`

**CLI Type**
Antigravity (Google OAuth via cloudcode-pa.googleapis.com)

**Model Name**
All models via Antigravity provider (gemini-claude-*, gemini-2.5-*, etc.)

**LLM Client**
Claude Code

**Request Information**
This is not a request issue. It's a background refresh loop issue. Log evidence:
```
INFO auth file changed: antigravity-account_a.json
INFO auth file changed: antigravity-account_b.json
INFO auth file changed: antigravity-account_c.json
INFO auth file changed: antigravity-account_d.json
INFO auth file changed: antigravity-account_e.json
INFO auth file changed: antigravity-account_a.json  // ← Same account again after ~5 seconds!
```
Each account refreshes every ~5-6 seconds continuously.

**Expected behavior**
- Token should be refreshed only when approaching expiry (e.g., 5 minutes before)
- After successful refresh, `expired`, `timestamp`, and `access_token` should be persisted to disk
- Next refresh check should see the updated expiry time and skip refresh

**OS Type**
- OS: macOS
- Version: Darwin 25.1.0

**Additional context**

Related constants that affect refresh behavior:
- `refreshCheckInterval = 5 * time.Second` (conductor.go:42)
- `RefreshLead = 5 * time.Minute` (antigravity.go:46-48)
- `refreshSkew = 3000 * time.Second` (antigravity_executor.go:47) - This is 50 minutes, quite aggressive

The original intent of `metadataEqualIgnoringTimestamps` was to prevent fsnotify refresh loops, but it inadvertently caused a much worse refresh loop by not persisting token updates.

### PR DESCRIPTION
**Pull request by moremorefun**
## Context

Two related issues were causing problems with Antigravity OAuth token management:

1. **Infinite refresh loop (#833)**: `metadataEqualIgnoringTimestamps()` ignored `access_token`, causing refreshed tokens not to persist. The `expired` field remained stale, triggering refresh every 5 seconds.

2. **ModelStates loss (#797)**: When auth files were reloaded via fsnotify, `ModelStates` (containing `BackoffLevel` for 429 cooldown) was not preserved, resetting exponential backoff to 0.

## Changes

- **Token Persistence**: Removed `access_token` from ignored fields in `metadataEqualIgnoringTimestamps()` to ensure refreshed tokens are persisted to disk.

- **Self-Write Detection**: Added `ExpectedWriteTracker` using SHA256 content hashing to skip self-triggered fsnotify events, preventing event storms.

- **File-Triggered Updates**: Introduced `UpdateFromFileChange()` in conductor that updates memory state without writing back to file, breaking the event loop.

- **ModelStates Preservation**: Added deep copy of `ModelStates` in `Update()`, `UpdateFromFileChange()`, and `applyCoreAuthAddOrUpdate()` to preserve 429 backoff state across reloads.

- **429 RetryDelay Parsing**: Parse `retryDelay` from Antigravity 429 responses (e.g., `"10627.493230411s"`) to use upstream-specified cooldown instead of default exponential backoff.

- **Tests**: Added 249 lines of unit tests for `parseAntigravityRetryDelay` covering edge cases.

## Fixes

Fixes #833
Fixes #797

### CODE REVIEW

File: internal/runtime/executor/antigravity_executor_test.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+				{"@type": "type.googleapis.com/google.rpc.RetryInfo", "retryDelay": "-5s"}
+			]
+		}
+	}`)
+
+	result := parseAntigravityRetryDelay(body)
+	if result != nil {
+		t.Errorf("Expected nil for negative duration, got %v", *result)
+	}
+}
```
[the remark this answers is not in the document]
Comment (moremorefun) [reply internal/runtime/executor/antigravity_executor_test.go]: Done. Consolidated all negative test cases into a single table-driven test `TestParseAntigravityRetryDelay_NegativeCases` with 11 sub-tests. See commit 2f6f449.

File: internal/watcher/events.go [hunk 2]
```diff
@@ -124,6 +132,24 @@ func (w *Watcher) handleEvent(event fsnotify.Event) {
...
+// checkAndSkipSelfWrite reads the file and checks if content matches an expected write.
+// Returns true if this was a self-triggered write that should be skipped.
+func (w *Watcher) checkAndSkipSelfWrite(path string) bool {
+	if w.expectedWrites == nil {
+		return false
+	}
+	content, err := os.ReadFile(path)
+	if err != nil || len(content) == 0 {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (moremorefun) [reply internal/watcher/events.go:151]: Done. Added debug log when `os.ReadFile` fails in `checkAndSkipSelfWrite`. See commit 2f6f449.

File: sdk/cliproxy/auth/conductor.go [hunk 3]
```diff
@@ -224,9 +224,19 @@ func (m *Manager) Update(ctx context.Context, auth *Auth) (*Auth, error) {
...
+			auth.indexAssigned = existing.indexAssigned
+		}
+		// Preserve runtime state that is not persisted (deep copy)
+		if len(auth.ModelStates) == 0 && len(existing.ModelStates) > 0 {
+			copiedStates := make(map[string]*ModelState, len(existing.ModelStates))
+			for k, v := range existing.ModelStates {
+				copiedStates[k] = v.Clone()
+			}
+			auth.ModelStates = copiedStates
+		}
```
[the remark this answers is not in the document]
Comment (moremorefun) [reply sdk/cliproxy/auth/conductor.go]: Done. Extracted `CloneModelStates` helper function in `sdk/cliproxy/auth/types.go` and updated all three call sites. See commit 2f6f449.

File: internal/watcher/events.go [hunk 4]
```diff
@@ -124,6 +135,28 @@ func (w *Watcher) handleEvent(event fsnotify.Event) {
...
+	content, err := os.ReadFile(path)
+	if err != nil {
+		log.Debugf("watcher: failed to read file for self-write check on %s: %v", filepath.Base(path), err)
+		return false
+	}
+	if len(content) == 0 {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (moremorefun) [reply internal/watcher/events.go:151]: @gemini-code-assist This check is intentionally kept as a defensive measure.

**Why empty files are not expected from our writes:**
- In `filestore.go`, the `Save` method uses `json.Marshal(auth.Metadata)` which produces at minimum `{}` for a non-nil map
- Therefore, `ExpectWrite()` is never called with empty content, so this scenario cannot occur from self-writes

**Why we skip empty files:**
- Empty files indicate corruption, interrupted writes, or transient states during atomic operations
- Processing them would likely cause JSON parsing errors downstream
- Auth files are always valid JSON, so empty content signals an abnormal state that should not trigger client updates

The perceived inconsistency (registering empty hash but not checking) does not occur in practice because filestore never writes empty files.
